### DISCUSSION TITLE
Help understanding ESPHome config

### DISCUSSION THREAD

**Discussion by umtauscher** [echavet/MitsubishiCN105ESPHome#67] [Q&A]
Hi,
I have very great difficulty understanding what parts are ESPHome config bound and what parts are to be changed programatically.

I have an example showing two different  UI controls of the same Mitsubishi airconditioner.
The one titled "Klima Ruths Büro" is the standard control generated ny the Melcloud hacs component.
The one tiled "Klima" is the standard control generated by the MisubishiCN105ESPHome component.

As you can see, the latter is showing only swing modes, and the former ist showing settings for the vertical vane.

What I am asking, how could I implement the vertical vane settings in the ESPHome component and even better adding the horizontal van as well. Additionally I would like to add an auto mode to the horizontal settings, as the airconditioner supports it.

Perhaps anybody could point me in the right direction.
TIA
Wilhelm   
![Bildschirmfoto 2024-03-22 um 11 32 45](https://github.com/echavet/MitsubishiCN105ESPHome/assets/24495410/4c865c48-27b1-434a-b90d-a013f7dd0101)
![Bildschirmfoto 2024-03-22 um 11 34 28](https://github.com/echavet/MitsubishiCN105ESPHome/assets/24495410/17b9e28f-a0cf-493d-8d69-556003857816)

**Comment by supermarkert** [comment]
I think I have a similar question. My unit only has a Vertical vane swing option, so that's really all I want to see. I tried removing the "Horizontal" line of the config, but I still get all the options. Is this correct way to configure?
```climate:
  - platform: cn105
    id: hp
    name: "${friendly_name}"
    icon: mdi:heat-pump
    visual:
      min_temperature: 16
      max_temperature: 31
      temperature_step:
        target_temperature: 1
        current_temperature: 0.1
    compressor_frequency_sensor:
      name: ${name} Compressor Frequency
    vertical_vane_select:
      name: ${name} Vertical Vane
    remote_temperature_timeout: 30min
    debounce_delay : 500ms
    update_interval: 3s
```

![Screenshot 2024-03-24 102528](https://github.com/echavet/MitsubishiCN105ESPHome/assets/15205204/640f429b-ed80-4e10-8d84-a21e07b71509)

**Comment by disruptivepatternmaterial** [comment]
It is my understanding that this is a limitation of home assistant, if you look at the config options there  (https://www.home-assistant.io/integrations/climate/) it only has climate.set_aux_heat, climate.set_preset_mode, climate.set_temperature, climate.set_humidity, climate.set_fan_mode, climate.set_hvac_mode, climate.set_swing_mode, climate.turn_on, climate.turn_off, climate.toggle so we really need 2 swing modes, but there is only one.

So what I do is I have an automation that just keeps my vertical only units in vertical and my dual units in both mode. As far as I can tell, it is hard to see what the unit supports, of that data were there we could only show supported swings, but I am not seeing it in the data...but I could be missing something!

**Reply by umtauscher** [reply]
Swing mode has a numeric parameter. Unfortunately this parameter is not further specified, but the standard climate control of Homeasissistant has auto 1-5 and swing which could be 0-6 for the parameter

On my unit using the Melcloud integration this steers the vertical vane from 1 up to 5 down ant swing. I suppose the auto setting is unclear to me.

**Comment by echavet** [comment]
Hi @umtauscher,
thank you for your message. 
Your capture clearly shows the first component is able to customize/override the Climate Component swing labels.
I must admit I don't know how it is done. I thought it couldn't be done without customizing the HA UI Climate component. 
This is perhaps what MelCloud dev did. I can see the integration comes from hacs which confirms it could be the case.

**Comment by umtauscher** [comment]
Mitsubishi2mqtt has the Swing settings as in the first screenshot, but it‘s also not ESPHome.

**Comment by echavet** [comment]
You're right, I just checked the code and it is clearly related to esphome Climate implementation. So I imagine there is a way to override it.
